std/experimental/allocator/building_blocks/bitmapped_block.d

```diff
@@ -1746,3 +1760,344 @@ private struct BitVector
+    an appropriate contiguous hunk of memory. Regardless of the constructor
+    used, the destructor releases the memory by using `ParentAllocator.deallocate`.)
+    )
+    */
+    this(ubyte[] data)
+    {
+        immutable a = data.ptr.effectiveAlignment;
+        assert(a >= size_t.alignof || !data.ptr,
+            "Data must be aligned properly");
+
+        immutable ulong totalBits = data.length * 8;
+        immutable ulong bitsPerBlock = blockSize * 8 + 1;
+        _blocks = totalBits / bitsPerBlock;
+
+        // Reality is a bit more complicated, iterate until a good number of
+        // blocks found.
+        size_t localBlocks;
+        for (localBlocks = _blocks; localBlocks; --localBlocks)
+        {
+            immutable controlWords = localBlocks.divideRoundUp(64);
+            auto payload = data[controlWords * 8 .. $].roundStartToMultipleOf(
+                alignment);
+            if (payload.length < localBlocks * blockSize)
+            {
+                // Overestimated
+                continue;
+            }
+            _control = (cast(shared(ulong*)) data.ptr)[0 .. controlWords];
+            _control[] = 0;
+            _payload = cast(typeof(_payload)) payload;
+            break;
+        }
+        _blocks = cast(typeof(_blocks)) localBlocks;
+    }
+
+    /// Ditto
+    static if (chooseAtRuntime == theBlockSize)
+    this(ubyte[] data, uint blockSize)
+    {
+        this._blockSize = blockSize;
+        this(data);
+    }
+
+    /// Ditto
+    static if (!is(ParentAllocator == NullAllocator))
+    this(size_t capacity)
+    {
+        size_t toAllocate = totalAllocation(capacity);
+        auto data = cast(ubyte[]) (parent.allocate(toAllocate));
+        this(data);
+        assert(_blocks * blockSize >= capacity);
+    }
+
+    /// Ditto
+    static if (!is(ParentAllocator == NullAllocator) &&
+        chooseAtRuntime == theBlockSize)
+    this(size_t capacity, uint blockSize)
+    {
+        this._blockSize = blockSize;
+        this(capacity);
+    }
+
+    /**
+    Returns the actual bytes allocated when `n` bytes are requested, i.e.
+    `n.roundUpToMultipleOf(blockSize)`.
+    */
+    pure nothrow @safe @nogc
+    size_t goodAllocSize(size_t n)
+    {
+        return n.roundUpToMultipleOf(blockSize);
+    }
+
+    /**
+    Allocates `s` bytes of memory and returns it, or `null` if memory
+    could not be allocated.
+
+    The `SharedBitmappedBlock` cannot allocate more than the given block size.
+    Allocations are satisfied by searching the first unset bit in the bitmap,
+    and atomically setting it.
+    In rare memory pressure scenarios, the allocation could fail.
+    */
+    @trusted void[] allocate(const size_t s)
+    {
+        import core.atomic : cas, atomicLoad, atomicOp;
+
+        if (s.divideRoundUp(blockSize) != 1)
+            return null;
+
+        // First zero bit position for all values in the 0 - 255 range
+        // for fast lookup
+        static ubyte[] firstZero = [
```

Comment:
Make this `static immutable ubyte[256] firstZero = [ ... ];` so it goes into the rodata section. Also initializing with ctfe is nice and won't harm.

Reply:
To initialize with ctfe:

```d
static ubyte[256] initFirstZero()
{
    ... do it ...
}
static immutable ubyte[256] firstZero = initFirstZero();
```